Automattic/woocommerce-payments
# Add WCPay Subscriptions empty state screens

Fixes #3281 

#### Changes proposed in this Pull Request

This PR adds the empty states for WCPay Subscriptions stores when they have no customer subscriptions yet. 

### Unconnected account view

<img width="2560" alt="Screen Shot 2021-11-22 at 4 26 00 pm" src="https://user-images.githubusercontent.com/8490476/142810806-73e70740-03bb-4d32-a108-5ffc86def5a9.png">

### Connected account view 

<img width="2558" alt="Screen Shot 2021-11-22 at 4 24 21 pm" src="https://user-images.githubusercontent.com/8490476/142810642-fc48a160-3e25-4bd2-b5df-0cb34923863d.png">

**Note**: This is my first real attempt at writing react. I've mostly just tried to look at what others have done and emulate their approach. Let me know if I've done something incorrectly. :) 

From the designs (jcP4y1b7zYi6PfqX8Q22qr-fi-521%3A61308) the following things are missing 

- [x] This was fixed with 94744a1 ~When you click any of the buttons they don't get the `isBusy` styling that is used elsewhere. I tried using to add that using the `useState` function similar to how it has been done, however I was getting console errors. If you could show me how to do that, that would be awesome.~
   - Here's what I tried https://www.diffchecker.com/XEkodIjx
   - [Here's the error I was getting](https://user-images.githubusercontent.com/8490476/142820922-a93e8a64-c2fb-46c0-9987-17e4299e935d.png)
   - I had a look at [this post](https://reactjs.org/warnings/invalid-hook-call-warning.html) on the subject but couldn't work out what I need to change.
- [ ] After clicking "Finish setup" and completing the KYC, the user isn't redirected back to the **WooCommerce > Subscriptions** screen. I'm not sure how to do that. 
- [x] This was fixed in 9f5ed72 ~When you click the create subscription product button, the screenshot in the designs suggest that 
**"Simple subscription"** product type should be selected, we will need further changes to implement that.~


#### Testing instructions

1. Get a fresh site with no subscriptions. You can either: 
   - Create a fresh JN site and install WooCommerce and this branch of WC Payments (see built file zip below). Or
   - Go into your local site's database and run a query like the one below to temporarily remove all subscriptions.
2. Go to **WooCommerce > Subscriptions** in your WP dashboard and you should see one of the screens above based on your WC Payments connected status. 
3. Connect or disconnect your account and observe both screens.

**"Delete" all subscriptions** 
```sql
UPDATE `wp_posts`
SET `post_type` = 'subscription_backup'
WHERE `post_type` = 'shop_subscription';
```

**Bring back all subscriptions** 
```sql
UPDATE `wp_posts`
SET `post_type` = 'shop_subscription'
WHERE `post_type` = 'subscription_backup';
```

**Built files**
[woocommerce-payments.zip](https://github.com/Automattic/woocommerce-payments/files/7578801/woocommerce-payments.zip)

-------------------


**Post merge**
